lib/src/main/kotlin/com/swmansion/starknet/data/serializers/TransactionSerializer.kt

```diff
@@ -0,0 +1,92 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.data.types.*
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+internal val transactionIgnoredKeys = listOf("transaction_hash", "contract_address")
```

Comment:
These fields are not in RPC spec, so we remove them before while serializing.

Reply:
are they returned by nodes? or what is the reason for this disparity?

Reply:
You can see in `Transaction.kt` file that there are comments about these fields (e.g. [here](https://github.com/software-mansion/starknet-jvm/blob/7b712699a626fc51e9ce665dbb539c57d86c798a/lib/src/main/kotlin/com/swmansion/starknet/data/types/Transaction.kt#L694)). I'm not sure about the specific reason for this disparity, but these fields **are clearly not in the spec.**

E.g., here's deploy account v1 as [kotlin dataclass](https://github.com/software-mansion/starknet-jvm/blob/7b712699a626fc51e9ce665dbb539c57d86c798a/lib/src/main/kotlin/com/swmansion/starknet/data/types/Transaction.kt#L757) and as [spec schema](https://github.com/software-mansion/starknet-jvm/blob/7b712699a626fc51e9ce665dbb539c57d86c798a/lib/src/main/kotlin/com/swmansion/starknet/data/types/Transaction.kt#L757), you can see the difference.

Reply:
> if they are missing from response, do not fail but just set them to something generic like null or something

Sure, I made contract address optional in `DeployAccountTransaction` + its inheriting classes and in `L1HandlerTransaction`. In `InvokeTransactionV0` it should stay according to spec. Transaction hash is already optional.

> let's reach out to relevant teams to ask about this disparity (maybe those fields should be added to the spec? or maybe nodes should remove them?)

Sure, will contact Ariel

> btw, is this for all the nodes? or is this only for either one of pathfinder/juno?

I know this happens on devnet's node and juno's node (the one that we utilize internally).